minedd/embeddings.py

```diff
@@ -0,0 +1,169 @@
+    This class has an internal docs object that is used to store a list of PaperQA Doc objects.
+    Each Doc object contains internally the embeddings of chunks of a single paper plus metadata with provenance (n_docs == n_processed_papers).
+
+    Attributes:
+        model (str): Generative Model that later will be used for answering queries.
+        embedding_model (str): Encoder Model to use for creating the embeddings.
+        output_embeddings_path (str): Path to save the output embeddings file.
+        paper_directory (str): Directory containing the papers to process.
+        docs (Docs, optional): Existing Docs object, to be able to manipulate it at Runtime.
+    """
+    def __init__(self,
+                 model: str = "ollama/llama3.2:1b",
+                 embedding_model: str = "ollama/mxbai-embed-large:latest",
+                 output_embeddings_path: str = "out/embeddings.pkl",
+                 paper_directory: str = "data/",
+                 existing_docs: Docs = None,
+                 ):
+        self.output_embeddings_path = output_embeddings_path
+        self.paper_directory = paper_directory
+        self.embedding_model = embedding_model
+        self.docs = existing_docs
+
+        self.settings = configure_settings(model, self.embedding_model, self.paper_directory)
+
+        emb_directory = os.path.dirname(self.output_embeddings_path)
+        if emb_directory:
+            os.makedirs(emb_directory, exist_ok=True)
+
+
+    def __repr__(self):
+        return f"Embeddings(output_embeddings_path={self.output_embeddings_path}, docs={len(self.docs)})"
+
+    def load_existing_embeddings(self, embeddings_path: str):
+        """Load existing embeddings (List of Docs) from the specified path."""
+        if os.path.exists(embeddings_path):
+            self.docs = safely_load_pickle_file(embeddings_path)
+            self.output_embeddings_path = embeddings_path
+            print(f"Loaded existing embeddings from {embeddings_path}")
+        else:
+            print(f"No existing embeddings found at {self.output_embeddings_path}")
+            self.docs = None
+
+
+    def get_docs_details(self, verbose=False):
+        """Get the details of the documents in the embeddings.
+        This method retrieves the metadata of the documents stored in the embeddings.
+        It returns a DataFrame with the details of each document.
+        If verbose is True, it also prints the details of each document. 
+        """
+        def get_safe_key(doc_detail, key):
+            try:
+                return getattr(doc_detail, key)
+            except AttributeError:
+                return None
+
+        if self.docs is None:
+            print("No existing embeddings found.")
+            return {}
+
+        doc_details = []
+        for doc_key in self.docs.docs.keys():
+            doc_detail = self.docs.docs.get(doc_key)
+            if doc_detail:
```

Comment:
This part here retrieves metadata and stores them in a very neat way. Well done!